materialize-elasticsearch/driver.go

```diff
@@ -96,27 +170,27 @@ func (driver) Validate(ctx context.Context, req *pm.Request_Validate) (*pm.Respo
 		return nil, fmt.Errorf("parsing endpoint config: %w", err)
 	}
 
-	var elasticSearch, err = newElasticSearch(cfg.Endpoint, cfg.Username, cfg.Password)
+	var elasticSearch, err = newElasticsearchClient(cfg)
 	if err != nil {
 		return nil, fmt.Errorf("creating elasticSearch: %w", err)
 	}
 
 	var out []*pm.Response_Validated_Binding
-	for _, binding := range req.Bindings {
+	for i, binding := range req.Bindings {
 		var res resource
 		if err := pf.UnmarshalStrict(binding.ResourceConfigJson, &res); err != nil {
 			return nil, fmt.Errorf("parsing resource config: %w", err)
 		}
 
 		// Make sure the specified resource is valid to build
-		if schema, err := schemabuilder.RunSchemaBuilder(
-			binding.Collection.GetReadSchemaJson(),
-			res.FieldOverides,
-		); err != nil {
-			return nil, fmt.Errorf("building elastic search schema: %w", err)
-		} else if err = elasticSearch.CreateIndex(res.Index, res.NumOfShards, res.NumOfReplicas, schema, true); err != nil {
+		if desc, err := elasticSearch.ApplyIndex(res.Index, res.NumOfShards, res.NumOfReplicas, true); err != nil {
 			// Dry run the index creation to make sure the specifications of the index are consistent with the existing one, if any.
 			return nil, fmt.Errorf("validate elastic search index: %w", err)
+		} else {
+			log.WithFields(log.Fields{
```

Comment:
I'm not 100% sure what to expect from this log message, but if it seemed useful to you to keep here then I'm good with that.

Reply:
yeah, I wanted to do something with the action description that's output from the `ApplyIndex` during validation. Many of the error messages in this connector lack details about which binding they relate to, and this just seemed like an easy way to identify that a particular binding was validated without issue.